src/mtcp/mtcp_restart.c

```diff
@@ -1262,10 +1273,32 @@ static void lock_file(int fd, char* name, short l_type)
   }
 }
 
+static char* fix_parent_directories(char* filename) {
+  int mtcp_sys_errno;
+  int i;
+  for (i=1; filename[i] != '\0'; i++) {
+    if (filename[i] == '/') {
+      int rc;
+      filename[i] = '\0';
+      rc = mtcp_sys_mkdir(filename, S_IRWXU);
+      if (rc < 0 && mtcp_sys_errno != EEXIST ) {
+        MTCP_PRINTF("error %d re-creating directory %s\n",
+		    mtcp_sys_errno, filename);
+        filename[i] = '/';  // Restore original filename
+	mtcp_abort();
+      } // else parent directory already exists or was re-created
+      // FIXME:  We should unlink parent directories if we re-created them.
+      filename[i] = '/';  // Restore original filename
+    }
+  }
+  return filename;  // Success.  All parent directories now exist.
+}
+
 #if 0
 /* Adjust for new pwd, and recreate any missing subdirectories. */
 static char* fix_filename_if_new_cwd(char* filename)
 {
+  int mtcp_sys_errno;
```

Comment:
If we do change this code, it should be a separate commit, of course.  For now, I propose we simply add a FIXME comment for this fnc:  /* Could be re-written to use fix_parent_directories in both parts of the function */
The code is essentially, try to recreate old directories;  If we fail, then chop off the prefix for which there really is such a directory, and treat the rest as a relative pathname.  Then try to create the relative pathname.  Both the first half and the second half (with a relative pathname) could be made to use fix_parent_directories().  Note, by the way, in that case fix_parent_directories() really should be revised to return NULL on failure, rather than abort on failure.